Fix Docker image build: add buf generate step to Dockerfile

## Problem

The Docker image build (`Build and publish gateway image` workflow) has been failing since March 10 with:

```
internal/filesclient/client.go:11:2: no required module provides package github.com/agynio/gateway/gen/agynio/api/files/v1
```

The `gen/` directory is gitignored and generated by `buf generate` in CI, but the Dockerfile doesn't run this step before `go build`.

## Solution

Add a `buf` installation stage and `buf generate` step to the Dockerfile so it's self-contained.

### Changes

1. **Dockerfile** — Add a `buf` download stage, copy `buf` into the build stage, run `buf generate buf.build/agynio/api --path agynio/api/files/v1` before `go build`. Place the generate step after `go mod download` and before `COPY . .` for optimal layer caching.

2. **`.dockerignore`** — Add `gen/` to prevent stale local generated files from being included in the build context.

### Key details
- `buf.gen.yaml` uses **remote plugins** — no `protoc` installation needed, only the `buf` CLI
- `buf` binary is ~30MB static download from GitHub Releases
- No `buf.lock` file exists in the repo currently
- Layer ordering: `go mod download` → `buf generate` → `COPY . .` → `go build`

--- a/.dockerignore
+++ b/.dockerignore
@@ -3,6 +3,7 @@
 charts
 dist
 .openapi
+gen/
 *.log
 node_modules
 tmp

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,41 +1,42 @@
 # syntax=docker/dockerfile:1.7
-
 ARG GO_VERSION=1.22
+ARG BUF_VERSION=1.64.0
 
+# Stage 1: Download buf binary
+FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-bookworm AS buf
+ARG BUF_VERSION
+RUN curl -sSL \
+      "https://github.com/bufbuild/buf/releases/download/v${BUF_VERSION}/buf-$(uname -s)-$(uname -m)" \
+      -o /usr/local/bin/buf && \
+    chmod +x /usr/local/bin/buf
+
+# Stage 2: Generate + compile
 FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-bookworm AS build
-
 WORKDIR /src
+ENV CGO_ENABLED=0 GO111MODULE=on
 
-ENV CGO_ENABLED=0 \
-    GO111MODULE=on
+COPY --from=buf /usr/local/bin/buf /usr/local/bin/buf
 
 COPY go.mod go.sum ./
-
 RUN --mount=type=cache,target=/root/.cache/go-build \
     --mount=type=cache,target=/root/go/pkg/mod \
-    go mod download && \
-    go mod verify
+    go mod download && go mod verify
+
+COPY buf.gen.yaml ./
+RUN buf generate buf.build/agynio/api --path agynio/api/files/v1
 
 COPY . .
 
-ARG TARGETOS
-ARG TARGETARCH
-
+ARG TARGETOS TARGETARCH
 RUN --mount=type=cache,target=/root/.cache/go-build \
     --mount=type=cache,target=/root/go/pkg/mod \
     GOOS=$TARGETOS GOARCH=$TARGETARCH go build \
-      -trimpath \
-      -ldflags="-s -w" \
-      -o /out/gateway ./cmd/gateway
+      -trimpath -ldflags="-s -w" -o /out/gateway ./cmd/gateway
 
+# Stage 3: Runtime
 FROM gcr.io/distroless/static-debian12:nonroot
-
 WORKDIR /app
-
 LABEL org.opencontainers.image.source="https://github.com/agynio/gateway"
-
 COPY --from=build /out/gateway ./gateway
-
 EXPOSE 8080
-
 ENTRYPOINT ["/app/gateway"]

